frontend/views/components/Tooltip.vue

```diff
@@ -1,85 +1,150 @@
 <template lang='pug'>
 span.c-twrapper(
   tabindex='0'
+  @click='toggle'
   @mouseenter='show'
   @mouseleave='hide'
   @focus='show'
   @blur='hide'
   aria-label='text'
 )
   slot
+
+  .c-background(
+    v-if='(isActive || isVisible) && manual'
+    @click='toggle'
+    v-append-to-body=''
+  )
+
   .c-tooltip(
-    :style='stylesPosition'
-    :class='isTextCenter ? "has-text-center" : ""'
+    :style='styles'
+    :class='{"has-text-center": isTextCenter, "is-active": isActive}'
     v-if='isActive || isVisible'
     v-append-to-body=''
   )
+    modal-close(
+      v-if='manual'
+      @close='toggle'
+    )
+    // Default tooltip is text
     template(v-if='text') {{text}}
+    // But any content can fit in
     slot(v-else='' name='tooltip')
 </template>
 
 <script>
-/*
-NOTE: when needed, this component can be improved with more options:
-- Click instead of hover
-- More directions
-- Controlled tooltip from outside
-For now I've just did the needed API for this particular task but I think it's pretty easy to scale it.
-*/
+import { TABLET } from '@view-utils/breakpoints.js'
+import ModalClose from '@components/modal/ModalClose.vue'
+import trapFocus from '@utils/trapFocus.js'
+
 export default {
   name: 'Tooltip',
+  mixins: [trapFocus],
   props: {
     text: String,
     // Force to show tooltip manually
     isVisible: Boolean,
-    isTextCenter: Boolean,
+    manual: {
+      type: Boolean,
+      default: false
+    },
+    isTextCenter: {
+      type: Boolean,
+      default: false
+    },
     direction: {
       type: String,
-      validator: (value) => ['bottom', 'bottom-end', 'right', 'right-start', 'top'].includes(value),
+      validator: (value) => ['bottom', 'bottom-end', 'right', 'left', 'top', 'top-left'].includes(value),
       default: 'bottom'
+    },
+    opacity: {
+      type: Number,
+      required: false,
+      default: 0.95
     }
   },
+  components: {
+    ModalClose
+  },
   data: () => ({
     trigger: null,
     tooltip: null,
+    tooltipHeight: 0,
+    tooltipWidth: 0,
     isActive: false,
-    stylesPosition: null
+    styles: null
   }),
+  mounted () {
+    window.addEventListener('resize', this.adjustPosition)
+  },
+  beforeDestroy: function () {
+    window.removeEventListener('resize', this.adjustPosition)
+  },
   methods: {
     show () {
-      this.isActive = true
+      if (!this.manual) this.isActive = true
     },
     hide () {
-      this.isActive = false
+      if (!this.manual) this.isActive = false
+    },
+    toggle () {
+      this.isActive = !this.isActive
+      if (this.$slots.tooltip) {
+        document[this.isActive ? 'addEventListener' : 'removeEventListener']('keydown', this.trapFunction)
+        // Wait for element to be present before auto focus
+        setTimeout(() => this.focusOnFirst(this.$slots.tooltip[0].elm))
```

Comment:
This works great on first try but if we close the profile card and then open it again it doesn't focus anymore. 🐛